src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}, only {@code "true"}, {@code "false"} are allowed</li>
+ *                     <li>{@link Boolean}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}, if configured; otherwise {@code false}.</td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * @see MongoDialectConfigurator
+ */
+public final class MongoDialectSettings {
+    private final MongoClientSettings mongoClientSettings;
+    private final String databaseName;
+    private final boolean autoCommit;
+
+    private MongoDialectSettings(Builder builder) {
+        this.mongoClientSettings = builder.mongoClientSettingsBuilder.build();
+        this.databaseName = notNull("databaseName", builder.databaseName);
+        this.autoCommit = builder.autoCommit;
+    }
+
+    /**
+     * Gets the {@link MongoClientSettings} to be used when creating the internal {@link MongoClient}.
+     *
+     * @return The {@link MongoClientSettings}.
+     * @see Builder#applyToMongoClientSettings(Consumer)
+     */
+    public MongoClientSettings getMongoClientSettings() {
+        return mongoClientSettings;
+    }
+
+    /**
+     * Gets the name of a MongoDB database used as the {@linkplain Connection#getSchema() JDBC schema} of a
+     * {@linkplain Connection connection} {@linkplain MongoConnectionProvider#getConnection() obtained} from
+     * {@link MongoConnectionProvider}.
+     *
+     * @return The name of the default MongoDB database.
+     * @see Builder#databaseName(String)
+     * @see DatabaseMetaData#getSchemaTerm()
+     */
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    /**
+     * The {@linkplain Connection#getAutoCommit() auto-commit mode} of {@linkplain Connection connections}
+     * {@linkplain MongoConnectionProvider#getConnection() obtained} from {@link MongoConnectionProvider}.
+     *
+     * @return The default auto-commit mode.
+     * @see Builder#autoCommit(boolean)
+     */
+    public boolean isAutoCommit() {
+        return autoCommit;
+    }
+
+    /**
+     * Creates a new {@link MongoDialectSettings.Builder} based on {@code configProperties}.
+     *
+     * @param configProperties The {@linkplain Configurable#configure(Map) configuration properties}.
+     * @return A new {@link MongoDialectSettings.Builder}.
+     */
+    public static MongoDialectSettings.Builder builder(Map<String, Object> configProperties) {
+        return new MongoDialectSettings.Builder(notNull("configProperties", configProperties));
+    }
+
+    /** A builder for {@code MongoDialectSettings}. */
+    public static final class Builder {
+        private final MongoClientSettings.Builder mongoClientSettingsBuilder;
+        private @Nullable String databaseName;
+        private boolean autoCommit;
+
+        private Builder(Map<String, Object> configProperties) {
```

Comment:
In Hibernate codebase, usually the naming of the map is `configurationValues` or `configValues` for Hibernate does allow for ad-hoc value to be used as value, not restrictive to java `Properties`.

Reply:
Here is an ad-hoc config usage in Hibernate ORM codebase: https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/boot/cfgxml/internal/CfgXmlAccessServiceImpl.java

```
public class CfgXmlAccessServiceImpl implements CfgXmlAccessService {
	private final LoadedConfig aggregatedCfgXml;

	public CfgXmlAccessServiceImpl(Map<?,?> configurationValues) {
		aggregatedCfgXml = (LoadedConfig) configurationValues.get( LOADED_CONFIG_KEY );
	}

	@Override
	public LoadedConfig getAggregatedConfig() {
		return aggregatedCfgXml;
	}
}
```

`configProperties` is fine but it might mislead reader into thinking that it comes from `hibernate.properties` file exclusively. Actually it could be populated in ad-hoc way during Hibernate bootstrapping.

Reply:
Discussed. I think that `configurationValues` is wrong - that map contains configuration property names as keys, and configuration property values as values. Hibernate ORM API docs describe this map as [_"The configuration properties"_](https://docs.jboss.org/hibernate/orm/6.6/javadocs/org/hibernate/service/spi/Configurable.html#configure(java.util.Map)), which makes much more sense than `configurationValues`, which Hibernate ORM for some reason uses as the parameter name.

I renamed it nonetheless in fdc4c5feb1dff31b5751105ae5c9be4e83fc9ab0, because you want this.

Reply:
I think it is good to follow some convention; sometimes it is not technically accurate for historical reason but we have multiple perspectives to consider.